modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java

```diff
@@ -437,12 +437,19 @@ public Processor create(
                 // pipeline.
                 return new DatabaseUnavailableProcessor(processorTag, description, databaseFile);
             }
+
             final String databaseType;
             try {
                 databaseType = geoIpDatabase.getDatabaseType();
             } finally {
                 geoIpDatabase.release();
             }
+            if (databaseType == null
+                || (databaseType.endsWith(CITY_DB_SUFFIX)
+                    || databaseType.endsWith(COUNTRY_DB_SUFFIX)
+                    || databaseType.endsWith(ASN_DB_SUFFIX)) == false) {
+                throw newConfigurationException(TYPE, processorTag, "database_file", "Unsupported database type [" + databaseType + "]");
```

Comment:
Can databaseType actually be null here? In that case, do we want the output to be `Unsupported database type [null]` , or would a different message be good for that situation?

Reply:
I think we'd be hard-pressed to make it be `null` at runtime in the current code. But the code for loading the `databaseType` is funky enough even in the existing implementation (and that implementation could change in the future...), and indeed `GeoIpDatabase` is only an interface, so I think it's worth handling the possibility.

I added a separate before and after test for the null case, and I added the database file to the error message as well in both the null and the not-supported cases.

Reply:
I noticed that loading  code was indeed confusing! Adding the file name as additional context, in case of null type, looks good